File: backend/src/models/University.ts
import {Schema,model,Document} from 'mongoose';
import Faculty from './Faculty';

const schema = new Schema({
    name: String,
    schools: Object,
}, {collection: 'universities'});

interface IUniversity extends Document {
    name: String;
    schools: Object;
}

export default model<IUniversity>('University',schema);

File: Unihub_app/lib/models/university.dart
class University {
  final String name;
  final List<dynamic> schools;

  University(
    this.name,
    this.schools,
  );

  factory University.fromMap(Map<String, dynamic> json) {
    return University(
      json['name'],
      json['schools'],
    );
  }
}
